src/app/vm/vm-creation/vm-creation.component.ts

```diff
@@ -142,17 +143,22 @@ export class VmCreationComponent implements OnInit {
     this.enoughResources = true;
     this.resourceUsageService.getResourceUsage()
       .subscribe(resourceUsage => {
-        if (resourceUsage.available.cpus &&
-          resourceUsage.available.instances &&
-          resourceUsage.available.volumes
-        ) {
-          this.resetVmCreateData();
-        } else {
-          this.fetching = false;
+        Object.keys(resourceUsage.available)
+          .filter(key => key !== 'snapshots' && key !== 'secondaryStorage')
+          .forEach(key => {
+            const available = resourceUsage.available[key];
+            if (available === 0) {
+              this.insufficientResources.push(key);
```

Comment:
Please create a map where keys are Object.keys(resourceUsage.available) and values are messages not to use resource.toUpperCase() in template